### PR TITLE
Release R114 Polonnaruwa

### PR DESCRIPTION
**Pull request by benjben**

### CODE REVIEW

**Comment by benjben** [comment]
Good point for #3951 , I created #4078.

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala [hunk 1]
```diff
@@ -279,14 +282,20 @@ object EnrichmentManager {
...
+    // and if the user agent is defined.
     val iabContext = registry.getIabEnrichment match {
       case Some(iab) =>
-        iab
-          .getIabContext(Option(event.useragent),
-                         Option(event.user_ipaddress),
-                         Option(event.derived_tstamp).map(EventEnrichments.fromTimestamp))
-          .map(_.some)
+        event.user_ipaddress match {
+          case IPv4Regex(ipv4) if !List(null, "", s"\0").contains(event.useragent) =>
```
Comment (BenFradet) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala]:
```diff
-- s"\0"
++ "\0"
```

also are those checks necessary given the regex match?

also, I'd rather we do `Option(event.user_ipaddress).map ...`

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala [hunk 2]
```diff
@@ -42,7 +39,11 @@ import scalaz._
 import Scalaz._
 
 // Scala URI
-import com.netaporter.uri.Uri
+import io.lemonlabs.uri.Uri
+import io.lemonlabs.uri.Url
```
Comment (BenFradet) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/ConversionUtils.scala]: :art: we can combine those two imports

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala (hunk 1 above)
Comment (benjben) [reply 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala]: I'll update `"\0"`.

The regex is for the IP, while the checks are for the user agent.

Comment (benjben) [reply 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala]: I can't remove the `s` :
```
Octal escape literals are unsupported, use \u0000 instead
```

**Pull request merged**